File: Services/LogService.cs
using System;
using System.IO;

namespace InvolveX.Cli.Services
{
    public class LogService
    {
        private readonly string _logFilePath;

        public LogService()
        {
            var logDirectory = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "logs");
            if (!Directory.Exists(logDirectory))
            {
                Directory.CreateDirectory(logDirectory);
            }
            _logFilePath = Path.Combine(logDirectory, "update.log");
        }

        public void Log(string message)
        {
            try
            {
                File.AppendAllText(_logFilePath, $"{DateTime.Now}: {message}{Environment.NewLine}");
            }
            catch (Exception ex)
            {
                // Fallback: log to console if file logging fails
                Console.Error.WriteLine($"Error writing to log file: {ex.Message} - Message: {message}");
            }
        }
    }
}

File: services/LogService.js
const fs = require('fs');
const path = require('path');

class LogService {
    constructor(logDirectory = null) {
        if (logDirectory) {
            this.logFilePath = path.join(logDirectory, 'involvex.log');
        } else {
            this.logFilePath = path.join(__dirname, '..', 'logs', 'update.log');
        }

        // Ensure logs directory exists
        const logDir = path.dirname(this.logFilePath);
        if (!fs.existsSync(logDir)) {
            fs.mkdirSync(logDir, { recursive: true });
        }
    }

    log(message) {
        try {
            const timestamp = new Date().toISOString();
            const logEntry = `${timestamp}: ${message}\n`;
            fs.appendFileSync(this.logFilePath, logEntry);
        } catch {
            // If logging to file fails, do nothing.
            // Writing to the console here would corrupt the TUI display.
        }
    }
}

module.exports = LogService;
